sql/core/src/main/scala/org/apache/spark/sql/UDFRegistration.scala

```diff
@@ -144,11 +181,24 @@ class UDFRegistration private[sql] (functionRegistry: FunctionRegistry) extends
     val inputTypes = Try(ScalaReflection.schemaFor[A1].dataType :: Nil).toOption
     def builder(e: Seq[Expression]) = ScalaUDF(func, dataType, e, inputTypes.getOrElse(Nil), Some(name))
     functionRegistry.registerFunction(name, builder)
-    UserDefinedFunction(func, dataType, inputTypes)
+    UserDefinedFunction(func, dataType, inputTypes, Some(name))
   }
 
   /**
-   * Register a Scala closure of 2 arguments as user-defined function (UDF).
+   * Registers a Scala closure of 1 arguments as user-defined function (UDF).
+   * @tparam RT return type of UDF.
+   * @since 2.3.0
+   */
+  def register[RT: TypeTag, A1: TypeTag](name: String, func: Function1[A1, RT], deterministic : Boolean, distinctLike: Boolean): UserDefinedFunction = {
```

Comment:
After discussion with others, we decide to use the way you proposed. The related two PRs have been merged. Will make a change in this PR too. Thanks!